src/vendor/vectors.js

```diff
@@ -700,6 +701,33 @@ function toPairs(obj) {
   );
 }
 
+function* internalCombos(first, ...rest) {
+  if (rest.length === 0) {
+    for (const f of first) {
+      yield [f];
+    }
+    return;
+  }
+
+  for (const f of first) {
+    for (const [...r] of internalCombos(...rest)) {
+      yield [f, ...r];
+    }
+  }
+}
+
+/*
+ * return every combination, one from each array
+ */
+function combos(...many) {
+  return new ArrayWrapper(
+    function* outputGen() {
+      for (const t of internalCombos(...many)) yield [t];
+    },
+    { debug: false },
+  );
+}
+
```

Comment:
@gmierz the `combo` function is used to convert conjunctive normal form to disjunctive normal form.